### ISSUE TITLE
[BUG/ISSUE] Extremely low mean OH in Tropchem simulation versus Benchmark (v13.0.0 & v13.0.2)

### ISSUE THREAD

**Issue by Twize** [geoschem/geos-chem#732]
### Description of the problem
I am interested in running troposphere-only full-chem simulations with GEOS-Chem v13. However, for all of the simulations I have run thus far (both on our local machines here at the University of Toronto, and on AWS), I have found the mass-weighted mean OH to be less than half of that from the benchmark (~5 x 10^5 molec. cm-3 versus ~11 x 10^5 molec. cm-3 for the benchmark). The photolysis rates for O(1D) and O(3P) are also extremely reduced in the trop-chem simulation versus the benchmark. This issue leads to a large over-estimation in CO columns, as well as other species including O3. 

### Description of troubleshooting performed
I have tried multiple trop-chem simulations on both our local machine, as well as on AWS (using the GC v13.0.2 tutorial AMI as a template), and with various settings (i.e., MERRA-2 and GEOS-FP meteorology, 47 and 72 model levels, online and offline emissions) as well as the default un-modified settings when creating a run directory. In all cases, the mass-weighted mean OH given by metrics.py is less than half than that of the benchmark. I have also run benchmark simulations on both the UofT system and on AWS, and comparisons with the official benchmark show very minimal differences, and approximately the same mean OH. I have included a plot comparing the OH fields from the 1-month AWS simulations below: 

![OH_rst_compare_20190801_Default_vs_benchmark_AWS](https://user-images.githubusercontent.com/26906734/119031692-6022b200-b979-11eb-97a2-3d244fc4347a.jpg)

I have compared the J-values for O(1D) and O(3P) for the default trop-chem simulation and benchmark I ran on AWS. The photolysis rates for O(1D) are approximately an order of magnitude lower in the tropchem simulation (using default settings), and several orders of magnitude lower for O(3P). See comparison plots below: 

![O3O1D_Jvals_compare_20190716_AWS_benchmark_vs_tropchem](https://user-images.githubusercontent.com/26906734/119031910-9b24e580-b979-11eb-878a-3a03f304926d.jpg)
![O3O3P_Jvals_compare_20190716_AWS_benchmark_vs_tropchem](https://user-images.githubusercontent.com/26906734/119031921-9fe99980-b979-11eb-88cd-0c736de72ee3.jpg)

At this point, I am unsure whether it is a bug in the v13 GEOS-Chem codes, or with the the "default" settings that are provided when setting up a run directory, but the differences are apparent on both the systems I have tested, and I have run out of ideas as to what may be the cause. Any help would be appreciated! 

### GEOS-Chem version
version 13.0.0 (UofT system, git commit: b6f7aff35a42c798e3b7cc94c41aa669c27faccd)
version 13.0.2 (AWS, git commit: b36f8efcebc667e932a1d358ed497c1e88c87ded)

The AMI (which is based upon the 13.0.2 tutorial AMI) that I used for the AWS simulations is: 
ami-015aec90f03d8f2d3

### Description of modifications
No modifications. 

### Log files
Files below are for the tropchem simulation I ran on AWS (with default settings), had to change the file handle to .txt for some to be able to upload them.
 
- Run logs (if applicable):
[input.geos.txt](https://github.com/geoschem/geos-chem/files/6518221/input.geos.txt)
[HEMCO.log](https://github.com/geoschem/geos-chem/files/6518211/HEMCO.log)
[log.dryrun.txt](https://github.com/geoschem/geos-chem/files/6518217/log.dryrun.txt) 

### Software versions
- CMake version:
3.19.5 (UofT system) 
3.19.2 (AWS)

- Compilers (Intel or GNU, and version):
Intel compiler v19.0.3.199 (UofT system)
GNU Fortran compiler v13.0.2 (AWS)

- NetCDF version:
4.2 (UofT system) 
4.7.4 (AWS)

**Comment by yantosca** [comment]
Thanks for writing @Twize.  In 13.0.0 there was an issue that met fields were not being read in properly (see #700, #686, #688).  I am not sure if one of these is causing your issue.

The photolysis rates for O1D and O3P are I believe computed differently depending on whether or not you are using the full-atmosphere or troposphere only.  That might have some impact on your results.

I am tagging @msulprizio, who worked on the implementation of the trop-only UCX mechanism in GEOS-Chem, who may have more insight.

**Comment by Twize** [comment]
> Thanks for writing @Twize. In 13.0.0 there was an issue that met fields were not being read in properly (see #700, #686, #688). I am not sure if one of these is causing your issue.
> 
> The photolysis rates for O1D and O3P are I believe computed differently depending on whether or not you are using the full-atmosphere or troposphere only. That might have some impact on your results.
> 
> I am tagging @msulprizio, who worked on the implementation of the trop-only UCX mechanism in GEOS-Chem, who may have more insight.

Hi Bob, thanks for the quick reply! That bug you mention may possibly be contributing to the issue on our UofT machines, since I am running v13.0.0 there, but it was my understanding that the issue was remedied with v13.0.2 (which is the version I am running on AWS), so don't think it is the cause in that case. 

I could always try doing 1-month of a full-atmosphere simulation on AWS to see if the issue with the OH fields persists. I tried this on the UofT machines, and the difference in the mean mass-weighted OH was quite small between the simulations (still approximately half of the benchmark). 

Hopefully @msulprizio may have an inkling on what might be the cause of these issues!

**Comment by msulprizio** [comment]
Hi @Twize. I see you provided the GEOS-Chem dry run log file above. Could you also upload the GEOS-Chem log file for your production simulation(s)? I'm wondering if the restart file is not being found/read correctly.

**Comment by Twize** [comment]
Hi again @msulprizio, I am attaching the log file from the 4x5 1-month trop-chem run I did on AWS. Please let me know if there are any other files that you might need, or tests that I can run
 that may be useful in diagnosing the issue!  

[GC.log](https://github.com/geoschem/geos-chem/files/6547814/GC.log)

**Comment by yantosca** [comment]
@Twize: It does look like the met data are being read in properly -- you can see that the met fields are being read in to the end of the month (i.e. look for "Found all A1 met fields", etc.)

It also looks like the restart file input is good (as you can see that the values are mostly in the ppt to ppb range for most species (except for NRO2, NAP, SO4{H1,H2,H3,H4}).  To me, that indicates the restart file has been read OK.

**Comment by Twize** [comment]
> @Twize: It does look like the met data are being read in properly -- you can see that the met fields are being read in to the end of the month (i.e. look for "Found all A1 met fields", etc.)
> 
> It also looks like the restart file input is good (as you can see that the values are mostly in the ppt to ppb range for most species (except for NRO2, NAP, SO4{H1,H2,H3,H4}). To me, that indicates the restart file has been read OK.

@yantosca that's good, at least that hopefully rules out the input/met files as the cause of this issue. Let me know if there is anything else that you can think of that I can test which might give us more leads on the issue.

**Comment by Twize** [comment]
@yantosca @msulprizio, I ran another simulation on AWS, this time with troposphere + stratosphere (instead of trop-only), and the mean mass-weighted OH is much closer to the benchmark simulation in this run (~10.7 x 10^5 molec. cm-3 vs. ~4.9 x 10^5 molec. cm-3 in the trop-only simulation). So, it would appear that the issue is coming from something to do with the troposphere-only simulation in v13. 

I am attaching relevant files for this new run below:
[input.geos.txt](https://github.com/geoschem/geos-chem/files/6560908/input.geos.txt)
[HEMCO.log](https://github.com/geoschem/geos-chem/files/6560910/HEMCO.log)
[GC.log](https://github.com/geoschem/geos-chem/files/6560901/GC.log)
[log.dryrun.txt](https://github.com/geoschem/geos-chem/files/6560913/log.dryrun.txt)

I plan to plot up some of the fields from the restart file later today and compare against the benchmark and trop-only run, so I can upload those when they are ready.

**Comment by Twize** [comment]
Hi @yantosca @msulprizio, just following up with some plots of the differences in the restart files, as well as the J-values. 

In terms of the OH fields, here is the Benchmark (which I ran myself on AWS) compared with the strat+trop chem run (also on AWS): ![OH_rst_compare_20190801_AWS_benchmark_vs_fullchem](https://user-images.githubusercontent.com/26906734/120034611-4c0b3000-bfcb-11eb-94c6-59132877b266.jpg)

The OH concentrations are relatively close and where there are differences (max. difference on the order of **0.08 ppt**), they are quite inhomogeneous. Contrast this with the Benchmark versus the trop-chem only run (first figure I included in my original post). In that figure, the bias in the OH concentrations is strongly negative almost everywhere, with a maximum difference on the order of **~0.3 ppt**.

**********************************************

Plots of the O(1D) and O(3P) photolysis rates tell a similar story, with the benchmark and strat+trop chem simulation matching somewhat closely (at least on the same order of magnitude): 

![O3O1D_Jvals_compare_20190716_AWS_benchmark_vs_fullchem](https://user-images.githubusercontent.com/26906734/120036339-013ee780-bfce-11eb-8051-4a57e2e4bc3d.jpg)
![O3O3P_Jvals_compare_20190716_AWS_benchmark_vs_fullchem](https://user-images.githubusercontent.com/26906734/120036370-0b60e600-bfce-11eb-94a2-220a11d0c2a3.jpg)

Contrast these with the second and third figures from the original post, where the photolysis rates were significantly lower in the trop-chem only simulation, particularly for O(3P). 

**********************************************

Lastly, to illustrate the impact of the OH differences, I plotted the CO fields from the restart files at the end of the simulations. 
Here is the Benchmark versus the strat+trop chem run:

![CO_rst_compare_20190801_AWS_tropchem_vs_fullchem_(fixed_scale)](https://user-images.githubusercontent.com/26906734/120037385-8aa2e980-bfcf-11eb-8167-59af7ee5208a.jpg)
and then the benchmark versus the troposphere-only run:
![CO_rst_compare_20190801_AWS_benchmark_vs_tropchem_(fixed_scale)](https://user-images.githubusercontent.com/26906734/120037985-8d520e80-bfd0-11eb-9588-90cca41b2980.jpg)

The positive bias in the CO concentrations is significantly higher compared to the benchmark in the trop-chem only simulation versus the strat+trop chem simulation. 

In short, the differences between the benchmark and the strat+trop chem run versus the benchmark and the trop-chem only run are quite drastic, particularly in the photolysis rates and the mean OH. To me, this maybe points to an issue with the trop-chem only simulation is run in GC v13.

**Comment by r-pound** [comment]
I've been able to replicate this issue in version 13.1.0 with the model built with chemistry only in the troposphere (Tropchem OH average of ~4x10^5). Rebuilding the model with chemistry in troposphere and stratosphere (Standard OH average of ~10x10^5) resolves this so it does appear to be a Tropchem only issue

**Comment by mje1398** [comment]
@yantosca It would seem to me that we should pull TropChem? From what I remember it's no faster than standard and creates a problem as we don't benchmark it? If that's the case I don't really see an advantage to it?

**Comment by mje1398** [comment]
Thanks @yantosca. Perhaps we should remove the option to generate tropchem via the rundirectory as it doesn't work?

**Comment by Twize** [comment]
> I've been able to replicate this issue in version 13.1.0 with the model built with chemistry only in the troposphere (Tropchem OH average of ~4x10^5). Rebuilding the model with chemistry in troposphere and stratosphere (Standard OH average of ~10x10^5) resolves this so it does appear to be a Tropchem only issue

Glad to know that the issue is reproducible and wasn't only occurring on my end! Thus far, I have switched over to the trop + strat chem version of the model and have not had any issues since in my simulations.

**Comment by mje1398** [comment]
Glad you've found a solution. I'm thinking that we should just ditch the tropchem version as I don't think its being very useful.

**Comment by msulprizio** [comment]
>It would seem to me that we should pull TropChem? From what I remember it's no faster than standard and creates a problem as we don't benchmark it? If that's the case I don't really see an advantage to it?

About a year ago, the GCST tried to push for retiring the tropchem simulation. The GCSC agreed that we should have a single chemical mechanism but decided they wanted to keep the option for a reduced chemistry grid (see the [GCSC meeting minutes with that discussion here](http://acmg.seas.harvard.edu/geos/word_pdf_docs/steering_cmte/gc_sc_minutes_27May2020.pdf)). With that said, the tropchem simulation was never properly validated and as indicated by this discussion seems problematic. I suggest we bring up this topic again at the next GCSC meeting next month. I'll contact Randall and Daniel to put it on the agenda.

**Comment by mje1398** [comment]
Hi @msulprizio,

that’s great. From what I remember there was no time penalty for standard vs tropchem? I which case I can’t see a reason for keeping it?

mat

**Comment by msulprizio** [comment]
This was the breakdown of timing by component when I ran 1-month benchmarks for the standard (72 and 47 level) and tropchem (47 level) chemical mechanisms in GEOS-Chem 12.8.0:
<img width="598" alt="Screen Shot 2021-07-30 at 12 09 17 PM" src="https://user-images.githubusercontent.com/4805437/127681183-e4db6f36-6bdd-4410-917a-8903e0b01b5b.png">

**Comment by msulprizio** [comment]
This bug is now fixed in commit https://github.com/geoschem/geos-chem/commit/72f44cdceabf43cc1960b95e70961a0e46273470 off of GEOS-Chem 13.0.0. This commit has been tagged as [`13.0.0+TropchemFix`](https://github.com/geoschem/geos-chem/releases/tag/13.0.0%2BTropchemFix). Users of GEOS-Chem 13.0.0 - 13.2.0 can apply this change to their code using the following steps.

1. Navigate to the `GCClassic/src/GEOS-Chem/` directory. Note: Your path may differ if you cloned the [GCClassic wrapper](https://github.com/geoschem/GCClassic) and named it something other than GCClassic.
2. Do `git fetch -p` to obtain the latest state of the geoschem/geos-chem repository.
3. Create a new branch at the current state of your repository for merging the fix into using `git checkout -b bugfix/tropchem`.
4. Merge the bug fix into your new branch using `git merge 13.0.0+TropchemFix`.
5. Depending on your version, you may be prompted to resolve a conflict in `GeosCore/hco_interface_gc_mod.F90`. The conflicts should just be spacing differences (see likely conflict code below). I recommend taking the code in the HEAD state of the repository and removing  the text `, RXN_O3_2a`. Don't forget to remove the lines `<<<<<<< HEAD` and `=======` through `>>>>>>> 13.0.0+TropchemFix`

```
<<<<<<< HEAD
    USE FAST_JX_MOD,          ONLY : RXN_NO2, RXN_O3_1, RXN_O3_2a
    USE HCO_GeoTools_Mod,     ONLY : HCO_GetSUNCOS
    USE Input_Opt_Mod,        ONLY : OptInput
    USE State_Chm_Mod,        ONLY : ChmState
    USE State_Grid_Mod,       ONLY : GrdState
    USE State_Met_Mod,        ONLY : MetState
=======
    USE FAST_JX_MOD,      ONLY : RXN_NO2, RXN_O3_1
    USE HCO_GeoTools_Mod, ONLY : HCO_GetSUNCOS
    USE Input_Opt_Mod,    ONLY : OptInput
    USE State_Chm_Mod,    ONLY : ChmState
    USE State_Grid_Mod,   ONLY : GrdState
    USE State_Met_Mod,    ONLY : MetState
>>>>>>> 13.0.0+TropchemFix
```

6. Commit your merge in either the Git GUI (recommended) or via the command line.
7. Rebuild GEOS-Chem and re-run your tropchem simulations.

I have tested this fix in 13.2.0 with 1-month benchmark simulations. The model performance statistics and OH metrics are included below.

<img width="926" alt="Screen Shot 2021-09-02 at 9 17 53 PM" src="https://user-images.githubusercontent.com/4805437/131935967-355088d1-8c31-4f6d-ad37-b45f96c0b0aa.png">

![1-month Simulations with GEOS-Chem 13 2 0 + Tropchem Bug Fix](https://user-images.githubusercontent.com/4805437/131935638-30e8a175-efd5-4da9-90ec-dc2dec7aeb06.png)

This fix does not impact GCHP because GCHP does not support the troposphere-only chemistry option.

NOTE: In GEOS-Chem 13.2.1, we will be retiring the troposphere-only chemistry option following the recommendation of the GEOS-Chem Steering Committee. For more details. please see Github issue https://github.com/geoschem/geos-chem/issues/839.
